addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -542,6 +542,11 @@ def _unlink_if_correct_states(self):
         for holiday in self.filtered(lambda holiday: holiday.state not in ['draft', 'cancel', 'confirm']):
             raise UserError(_('You cannot delete an allocation request which is in %s state.') % (state_description_values.get(holiday.state),))
 
+    @api.ondelete(at_uninstall=False)
+    def _unlink_if_no_leaves(self):
+        if any(self.filtered(lambda allocation: allocation.holiday_status_id.requires_allocation == 'yes' and allocation.leaves_taken > 0)):
```

Comment:
This way `any()` will stop at the first matching occurence, as `filtered()` loops through all the records :-)

```suggestion
        if any(allocation.holiday_status_id.requires_allocation == 'yes' and allocation.leaves_taken > 0 for allocation in self):
```

Reply:
Indeed :) I thought about this first then I forget to change it, now it is done, thanks!